src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -213,6 +213,13 @@ public async Task<IReadOnlyList<JobInfo>> GetJobsByIdsAsync(byte queueType, long
             return await cmd.ExecuteReaderAsync(_sqlRetryService, JobInfoExtensions.LoadJobInfo, _logger, cancellationToken, "GetJobsByIdsAsync failed.");
         }
 
+        public async Task<IReadOnlyList<JobInfo>> GetActiveJobsByQueueTypeAsync(byte queueType, CancellationToken cancellationToken)
```

Comment:
I am not sure what the reason for adding this method is. If it just to prevent more than one bulk-update job running you should use forceOneActiveJobGroup on Enqueue method.